### PR TITLE
docs: add some general package docs for popular questions

### PR DESCRIPTION
**Pull request by codyoss**
- Details on ForceSendFields.
- How to cast an error for more details.
- Update README formatting and link to package docs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,59 +15,57 @@ and using:
...
-instead. These are the new and
-idiomatic Go libraries targeted specifically at Google Cloud Platform Services.
+If you're working with Google Cloud Platform APIs such as Datastore or Pub/Sub, consider using the
+[Cloud Client Libraries for Go](https://github.com/googleapis/google-cloud-go) instead. These are the new and idiomatic
+Go libraries targeted specifically at Google Cloud Platform Services.
 
-The generator itself and the code it produces are beta. Some APIs are
-alpha/beta, and indicated as such in the import path (e.g.,
-"google.golang.org/api/someapi/v1alpha").
+The generator itself and the code it produces are beta. Some APIs are alpha/beta, and indicated as such in the import
```
Comment (tbpg) [inline README.md:50]: Should we clarify what we mean by API here?

File: README.md [hunk 2]
```diff
@@ -15,59 +15,57 @@ and using:
...
 
-The Application Default Credentials authenticate as the application itself,
-which make them great for working with Google Cloud APIs like Storage or
-Datastore. They are the recommended form of authentication when building
-applications that run on Google Compute Engine or Google App Engine.
+The Application Default Credentials authenticate as the application itself, which make them great for working with
+Google Cloud APIs like Storage or Datastore. They are the recommended form of authentication when building applications
+that run on Google Compute Engine or Google App Engine.
 
 Default credentials are provided by the `golang.org/x/oauth2/google` package. To use them, add the following import:
```
Comment (tbpg) [inline README.md:61]: Adding the import is not enough -- what do you do with it?

File: README.md [hunk 3]
```diff
@@ -101,5 +99,9 @@ if err != nil {
 client := oauth2.NewClient(ctx, ts)
 ```
 
-See also: [golang.org/x/oauth2/google](https://godoc.org/golang.org/x/oauth2/google) package documentation.
+See also: [golang.org/x/oauth2/google](https://pkg.go.dev/golang.org/x/oauth2/google) package documentation.
+
+## More information
 
+For some more information related to all of the generated client please read through our
```
Comment (tbpg) [inline README.md:106]:
```suggestion
For some more information related to all of the generated clients please read through our
```

File: README.md [hunk 4]
```diff
@@ -15,59 +15,57 @@ and using:
...
 * For support, use the [golang-nuts](https://groups.google.com/group/golang-nuts) mailing list.
-* The code review instance may be found [here](https://code-review.googlesource.com).
 
 ## Status
+
 [![Go Reference](https://pkg.go.dev/badge/google.golang.org/api.svg)](https://pkg.go.dev/google.golang.org/api)
 
-These are auto-generated Go libraries from the Google Discovery Service's JSON description files of the available "new style" Google APIs.
+These are auto-generated Go libraries from the Google Discovery Service's JSON description files of the available "new
+style" Google APIs.
```
Comment (tbpg) [inline README.md:38]: I don't know what "new style" means.

File: README.md [hunk 5]
```diff
@@ -15,59 +15,57 @@ and using:
...
 
-Due to the auto-generated nature of this collection of libraries, complete APIs or specific versions can appear or go away without notice.
-As a result, you should always locally vendor any API(s) that your code relies upon.
+Due to the auto-generated nature of this collection of libraries, complete APIs or specific versions can appear or go
+away without notice. As a result, you should always locally vendor any API(s) that your code relies upon.
```
Comment (tbpg) [inline README.md:41]: Appear to or actually go away without notice?

Comment (tbpg) [inline README.md:41]: Should update this vendor advice for modules.

File: README.md [hunk 6]
```diff
@@ -15,59 +15,57 @@ and using:
...
-These client libraries are officially supported by Google.  However, the libraries are considered complete and are in maintenance mode. This means that we will address critical bugs and security issues but will not add any new features.
+These client libraries are officially supported by Google.  However, the libraries are considered complete and are in
+maintenance mode. This means that we will address critical bugs and security issues but will not add any new features.
 
-If you're working with Google Cloud Platform APIs such as Datastore or Pub/Sub,
-consider using the
-[Cloud Client Libraries for Go](https://github.com/googleapis/google-cloud-go)
-instead. These are the new and
-idiomatic Go libraries targeted specifically at Google Cloud Platform Services.
+If you're working with Google Cloud Platform APIs such as Datastore or Pub/Sub, consider using the
```
Comment (tbpg) [inline README.md:46]: Can we be more clear than "consider"? It's basically if the API exists over there, go use that one.

File: README.md [hunk 7]
```diff
@@ -15,59 +15,57 @@ and using:
...
-alpha/beta, and indicated as such in the import path (e.g.,
-"google.golang.org/api/someapi/v1alpha").
+The generator itself and the code it produces are beta. Some APIs are alpha/beta, and indicated as such in the import
+path (e.g., "google.golang.org/api/someapi/v1alpha").
 
 ## Application Default Credentials Example
 
-Application Default Credentials provide a simplified way to obtain credentials
-for authenticating with Google APIs.
+Application Default Credentials provide a simplified way to obtain credentials for authenticating with Google APIs.
```
Comment (tbpg) [inline README.md:55]: Without already knowing what ADC is (even if you do), this section is very confusing.

The important part is what a user needs to do to authenticate, right?

**Review by tbpg — CHANGES_REQUESTED** [review]
A lot of these comments are for existing content. So, feel free to follow up with a separate PR.

File: README.md (hunk 6 above)
Comment (codyoss) [reply README.md]: changed wording.

File: README.md (hunk 1 above)
Comment (codyoss) [reply README.md]: Switched to calling them clients.

File: README.md (hunk 7 above)
Comment (codyoss) [reply README.md]: RIght, and the hope really is they have to do nothing. I will try to update the examples to modernize them.

File: README.md [hunk 8]
```diff
@@ -15,91 +15,69 @@ and using:
...
+maintenance mode. This means that we will address critical bugs and security issues but will not add any new features.
 
-## Application Default Credentials Example
+If you're working with Google Cloud Platform APIs such as Datastore or Pub/Sub, please use the
+[Cloud Client Libraries for Go](https://github.com/googleapis/google-cloud-go) instead. These are the new and idiomatic
+Go libraries targeted specifically at Google Cloud Platform Services.
 
-Application Default Credentials provide a simplified way to obtain credentials
-for authenticating with Google APIs.
+The generator itself and the code it produces are beta. Some clients are in alpha/beta and indicated as such in the
```
Comment (tbpg) [inline README.md:50]: The code is beta but the client is alpha/beta?

File: README.md [hunk 9]
```diff
@@ -15,91 +15,69 @@ and using:
...
+## Authorization
 
-Default credentials are provided by the `golang.org/x/oauth2/google` package. To use them, add the following import:
+By default, each API will use [Google Application Default Credentials](https://developers.google.com/identity/protocols/application-default-credentials)
+for authorization credentials used in calling the API endpoints. This will allow your application to run in many
+environments without requiring explicit configuration.
 
 ```go
-import "golang.org/x/oauth2/google"
+client, err := storage.NewService(ctx)
```
Comment (tbpg) [inline README.md:60]: Probably don't use `storage` since that should be cloud.google.com/go/storage? Also, show the import?

File: doc.go [hunk 10]
```diff
@@ -51,4 +51,27 @@
...
+// feature by adding the field name to the `ForceSendFields` slice. See docs on
+// any struct for more details.
+//
+//
+// Inspecting errors
+//
+// All of the errors returned by a client's `Do` method may be cast to a
+// `*googleapis.Error`. This can be useful for getting more information on an
+// error while debugging.
+//  if _, err := svc.FooCall().Do(); err != nil {
```
Comment (tbpg) [inline doc.go:72]: Indent more so it shows up as code?

**Comment by tbpg** [comment]
Looking much better. Thank you!

File: doc.go (hunk 10 above)
Comment (codyoss) [reply doc.go:72]: I thought one one extra space was needed for this to work, does it take more?

File: README.md (hunk 8 above)
Comment (codyoss) [reply README.md]: Removed second line and moved this up to the section on stability.

File: README.md (hunk 9 above)
Comment (codyoss) [reply README.md]: Good ideas. Switch to sheets as workspace apis are only found in this repo.

**Review by tbpg — APPROVED** [review]
Thank you!

**Pull request merged**